netstandard/ref/System.cs

```diff
@@ -2522,6 +2522,18 @@ public partial interface IFormattable
     {
         string ToString(string format, System.IFormatProvider formatProvider);
     }
+    public readonly partial struct Index : System.IEquatable<System.Index>
```

Comment:
Could we extend the list of interfaces implemented to include `IComparable<System.Index>`

Reply:
I'm not sure we can define that behavior in a meaningful way. `Index` allows indexing from the end (`^1` in C#), but we don't know the upper bound. So how does, say, `5` compare to `^3`? Depending on the upper bound, `5` might be before or after `^3`. We could of course define that all values relative to start are sorted before all indexes relative to the end and that each is sorted by index (relative to end being descending of course). However, that be confusing. So I'd be in favor of not allowing this.